chore(color): conversion helper tool and file cleanup

JimB40 opened the pull request:
This PR
- integrates Figma as source for SVG graphics
- automated PNG build for color LCD radios via covert-gfx.sh

Summary of changes:

1. changed img-src files names & folder structure
- files follows naming <bmp_|mask_><type_><name>.svg scheme that follows Figma UI Design scheme
- bmp_ & mask_ prefix defines PNG type
- no subfolders to simplify PNG generation & review
- all SVG vector sources simplified end re-exported to img-src

2. Replaced PNG
- PNGs now have new names
- unused PNGs are removed
- conversion done using rsvg-convert for smaller size

3. Modified CMakeList.txt
- simplified

4. Modified *.cpp
- all *.cpp with *.lbm references are updated to reflect naming change
- _BuiltinIcon names left untouched to avoid conflicts (lots of changes in code)

5. tools script to convert SVG to PNG
- changed script name to convert-gfx.sh
- added --resvg param to force using resvg insted of rsvg-convert (testing)

Other PRs @pfeerick 
- this affects #6753 so probably this one should be merged first ?
- will close #6779 
- may close #6772 :)

JimB40 committed:
shell script to update all PNG for color LCD radios

JimB40 committed:
add generated CSV file

JimB40 committed:
code cleanup

JimB40 committed:
Main change & clean up

1. changed img-src files names & folder structure
- files follows naming <bmp_|mask_><type_><name>.svg scheme that follows Figma UI Design scheme
- bmp_ & mask_ prefix defines PNG type
- no subfolders to simplify PNG generation & review
- all SVG vector sources simplified end re-exported to img-src

2. Replaced PNG
- PNGs now have new names
- unused PNGs are removed
- conversion done using rsvg-convert for smaller size

3. Modified CMakeList.txt
- simplified

4. Modified *.cpp
- all *.cpp with *.lbm references are updated to reflect naming change
- _BuiltinIcon names left untouched to avoid conflicts (lots of changes in code)

5. tools script to convert SVG to PNG
- changed name to convert-gfx
- added --resvg param to force using resvg insted of rsvg-convert (testing)

JimB40 committed:
PNG & script

1. add missing Favourite PNGs for #6753

2. added --update parameter to script
- CSV now keeps track of SVG modification date
- running ./convert-gfx.sh --make all --update  will only re-genarate PNGs for SVG s with newer modification date that the one in CSV

JimB40 committed:
Radio Analogs PNG fix

JimB40 committed:
Delete update-csv-dates

JimB40 committed:
Delete unused PNG

JimB40 committed:
Delete unused SVG

pfeerick committed:
chore: remove old files

pfeerick committed:
chore: run `--update` and `--build`
Note: on MacOS

pfeerick commented:
Ok, so *finally* looking at this @JimB40 🤣 

I think you had doubled up on the CSV/script file... I think `tools/convert-gfx.sh` and `tools/convert-gfx-list.csv` were the old files? With `tools/png-radio-fw.sh` and `tools/png_radio_fw_list.csv` being the new?  

I've rebased this to make it merge-ready again (and hopefully didn't break something), removed those extra files, I then run `--update` which updated the CSV (looks like it had old file names/paths), and `--build` which generated smaller versions of several images. 

If everything looks ok to you, should be right to merge. If that old script was just my earlier one modified, don't think there is any need to keep that... better to not add two scripts doing basically the same job - any changes needed can be made to yours as time goes on ;)

philmoz commented on tools/png_radio_fw_list.csv:
```diff
@@ -0,0 +1,99 @@
+"file"; 320x240; 480x272; 800x480
```
I think this file should be in the bitmaps folder.

pfeerick committed:
revert: "chore: remove old files"

This reverts commit 3c30b4c88d88ea532fc0034e9dd21486f67bf4ad and deletes the other two scripts instead.

pfeerick committed:
chore: change from shell script to Python
OS agnostic and easier to maintain as the script gets more complex

pfeerick committed:
chore: add .DS_Store to .gitignore
Was going to add it during another PR, but here is as good as any place.

pfeerick committed:
feat: implement CSV regeneration for SVGs and PNGs in convert-gfx.py

pfeerick committed:
feat: update validation process for SVG and PNG files, update some PNGs

pfeerick committed:
feat: logging and progress indication

pfeerick committed:
feat: add dry-run option and orphaned PNG
Also remove ANSI formatting in preference to rich formatting and duration

pfeerick committed:
refactor: argument parsing with argparse

pfeerick committed:
fix: format CSV as CSV, not SSV
Also de-duplcate CSV handling and minor cleanups

pfeerick committed:
feat: add dependency check for required packages

pfeerick committed:
chore: remove old png_radio_fw

pfeerick committed:
fix: check for orphaned PNG files when validating and improve console output
Check for orphaned PNGs when validating rather than needing to use cleanup in `--dry-run` mode. SVGs are already checked against CSV so nothing more is needed there.
Also turn off rich's syntax highlighting as this was triggering number and keyword highlighting.

pfeerick committed:
feat: enhance cleanup functionality to remove unreferenced SVG files

pfeerick commented:
Thanks for that. Yes, my bad, the png_radio_fw just had some things convert_gfx, didn't, so added that in, and changed it to a Python script as it was already a bit long for a shell script (plus python lets you do a lot of platform-agnostic stuff in an easy to maintain manner). 

It now also (among other things I probably forgot):
- has nicer UI with progress bars and elapsed times :grin:
- verbose flag if you really need the file lists (`--verbose`)
- checks if there are any orphaned PNG files when validating (`validate png`, `validate all`)
- validation will now also tell you if there is an SVG that isn't in the CSV file (`validate svg`, `validate all`)
- allows you to delete PNG files that no longer have a matching SVG, and optionally SVG files not referenced the CSV (`cleanup`, `cleanup --include-svg`)
- allows for a dry run so you can see if any changes would be made by make or cleanup (`--dry-run`)
- allows you to update the CSV with new SVGs (`update-list`)

I'll stop fiddling with it now, and we can add to it more and make changes as time progresses.
